### PR TITLE
Revised the stackalloc operator article

### PR DESCRIPTION
**Pull request by pkulikov**
Contributes to #10741 
Supported by dotnet/samples#976

Also moved the article to the operators part of the C# reference.

### CODE REVIEW

File: docs/csharp/language-reference/operators/stackalloc.md
Comment (BillWagner) [inline docs/csharp/language-reference/operators/stackalloc.md]: See? We should mention this first. 😄

File: docs/csharp/language-reference/operators/stackalloc.md
Comment (BillWagner) [inline docs/csharp/language-reference/operators/stackalloc.md]: I think we should reverse the order of these types: Because `Span<T>` can be used in a safe context, we should encourage it.

**Review by BillWagner — CHANGES_REQUESTED** [review]
Thanks again @pkulikov 

I had one general comment on order of the examples. Other than that, and my question on the snippets, this is ready to go.

File: docs/csharp/language-reference/operators/stackalloc.md
Comment (pkulikov) [reply docs/csharp/language-reference/operators/stackalloc.md]: @BillWagner I agree with you: swapped the types. Thanks!

**Review by BillWagner — APPROVED** [review]
Thanks for making the updates @pkulikov 

I'll approve this, and once the samples are merged, and the build is clean, we can :shipit:

**Comment by BillWagner** [comment]
closing and reopening to get a fresh build.

**Pull request merged**
